### ISSUE TITLE
Can not reproduce the output

### ISSUE THREAD

**Issue by htrvu** [Rbrq03/ClassDiffusion#4]
Hello there.
I followed your provided code and data to train the concept at `data/dog`. However, I can not achieve as good output as showed in your paper. Am I missed something?

My outputs:

- Learing rate `1e-4`, batch size `1`, spl weight is `1.0`
![image](https://github.com/Rbrq03/ClassDiffusion/assets/77799783/16702da3-82a6-4552-9685-6fff4675f5cf)

- Learing rate `2e-5`, batch size `1`, spl weight is `1.0`(equivalent to learning rate `1e-5` and batch size `2` in your setting)
![image](https://github.com/Rbrq03/ClassDiffusion/assets/77799783/6323d2d9-1e9d-4a73-bb04-7f745d54e27f)

Training scripts:
```bash
export CLS_TOKEN="dog"
export MODEL_NAME="runwayml/stable-diffusion-v1-5"
export OUTPUT_DIR="./ckpt/${CLS_TOKEN}_cls_1e-4"
export INSTANCE_DIR="./data/${CLS_TOKEN}"

accelerate launch train_class_diffusion.py \
  --pretrained_model_name_or_path=$MODEL_NAME  \
  --instance_data_dir=$INSTANCE_DIR \
  --output_dir=$OUTPUT_DIR \
  --instance_prompt="a photo of a <new1> ${CLS_TOKEN}"  \
  --resolution=512  \
  --train_batch_size=1  \
  --learning_rate=1e-4  \
  --lr_warmup_steps=0 \
  --max_train_steps=500 \
  --scale_lr \
  --hflip  \
  --modifier_token "<new1>" \
  --no_safe_serialization \
  --use_spl \
  --spl_weight=1 \
  --cls_token "${CLS_TOKEN}" \
  --validation_steps=50 \
  --validation_prompt="a <new1> ${CLS_TOKEN} is swimming" \
  --tracker_name "custom-diffusion-multi"
```

**Comment by Rbrq03** [comment]
Hello @htrvu, Thanks for interest.
- For the first experimental settings, I think it won't produce a good result as you use a 10 times learning rate and tunning-based subject-driven generation are sensitive for the hyper-parameters. For more details you can see https://huggingface.co/docs/diffusers/training/dreambooth#script-parameters
https://huggingface.co/blog/dreambooth
They shows that tunning-based subject-driven generation(like dreambooth) will produce results of very different quality with even slight hyper-parameters change. 
- For the second experimental settings, we mentions that all experiment are conducted on 2 * Nvidia RTX4090 in our paper. So the learning rate should be 4e-5 in my environment by the argument `--scale_lr`. I think you can try to use this learning rate first considering the result you show seems don't fit to the custom dataset, though I am not sure if 4e-5 will be a proper learning rate for 1 gpu and batch 1.

The following are the results I produce in your caption.
![dog5](https://github.com/Rbrq03/ClassDiffusion/assets/115004993/8d353d91-d4da-482b-9d0f-e242523c0cb2)
![dog4](https://github.com/Rbrq03/ClassDiffusion/assets/115004993/a06b9608-ae21-41d4-b3ef-7e5577d1e974)
![dog3](https://github.com/Rbrq03/ClassDiffusion/assets/115004993/e713c56d-a76f-4838-9af5-b2c489a99ca3)
![dog2](https://github.com/Rbrq03/ClassDiffusion/assets/115004993/31f02795-08f5-4bc1-b882-17cada435870)
![dog1](https://github.com/Rbrq03/ClassDiffusion/assets/115004993/97cb77b7-9355-45c9-a1a4-89360746cc37)
![dog](https://github.com/Rbrq03/ClassDiffusion/assets/115004993/8ed95b1a-264f-4a1b-8252-3cff64210840)

If you continue to experience issues, feel free to reach out.

**Comment by htrvu** [comment]
Thanks for your quick reply. The results are better now!

**Comment by Rbrq03** [comment]
Glad to hear that!

**Comment by x2x5** [comment]
> Glad to hear that!

Hello,

I'm currently reading the experimental details in your paper and comparing them with the provided code. I've noted a few discrepancies and would appreciate some clarification:

Learning Rate: The paper specifies a learning rate of 1e-6, while the training script utilizes 1e-5. Could you please confirm the accurate learning rate used for the reported results?

Batch Size: The paper states a batch size of 2. Given that train_batch_size=2 in the script refers to the per-GPU setting, is it correct to infer that the global batch size, when using two GPUs, is effectively 4?

Finally, If I intend to train on a single GPU (as I found using 1 or 2 gpus have almost same speed), what specific hyperparameter adjustments would you recommend to ensure consistent training outcomes?

Thank you for your excellent work on this project!

| 超参数                       | 数值                           | 说明                                      |
| :--------------------------- | :----------------------------- | :---------------------------------------- |
| 基础模型                     | Stable Diffusion V1.5          | 所有实验和基线比较都基于此模型。          |
| 学习率                       | $10^{-6}$                      | 微调（fine-tuning）阶段使用的学习率。     |
| 批大小                       | 2                              | 微调时每个批次包含的样本数。              |
| 优化步数                     | 单个概念：500<br>多个概念：800 | 根据需要个性化的概念数量进行调整。        |
| 语义保持损失权重 ($λ_{spl}$) | 1.0                            | 在所有主要实验中使用的SPL损失权重。       |
| 推理指导尺度                 | 6.0                            | 图像生成（Inference）时使用的CFG Scale。  |
| 推理步数                     | 100                            | 图像生成（Inference）时采样器迭代的步数。 |
| 硬件                         | 2 × RTX 4090 GPUs              | 实验运行的硬件环境。                      |

**Comment by Rbrq03** [comment]
Hey @x2x5,
I am not sure if there is a typo in our paper, I will check it later and thank you for pointing that out. I tend to think that the hyperparameters in the script are what we use in our experiments.

For batch size, I think you are correct.

For hyperparameters in single GPU, I am not sure about that. I recommend that you can start with 4e-5 with batch size 1 or 2e-5 with batch size 2. But I am not sure if it's a proper hyperparameter. 

Let me know about any your further results and trials that you want to share (better to open a new issue)

**Comment by x2x5** [comment]
> Hey [@x2x5](https://github.com/x2x5), I am not sure if there is a typo in our paper, I will check it later and thank you for pointing that out. I tend to think that the hyperparameters in the script are what we use in our experiments.
> 
> For batch size, I think you are correct.
> 
> For hyperparameters in single GPU, I am not sure about that. I recommend that you can start with 4e-5 with batch size 1 or 2e-5 with batch size 2. But I am not sure if it's a proper hyperparameter.
> 
> Let me know about any your further results and trials that you want to share (better to open a new issue)

Thanks for your quick response~

The experiment setting in the paper is here. And that's why I feel a bit confused about which setting is correct.

<img width="1217" height="338" alt="Image" src="https://github.com/user-attachments/assets/b5ae5824-1819-4574-b308-953e7c7f1b0b" />
